plutus-scb/src/Cardano/Metadata/Types.hs

```diff
@@ -66,67 +82,148 @@ newtype PropertyName =
     deriving (Show, Eq, Generic)
     deriving newtype (ToJSON, FromJSON, Pretty)
 
-data Property =
-    Property
-        { _propertySubject     :: Subject
-        , _propertyDescription :: PropertyDescription
-        }
+type role SubjectProperties phantom
+
+data SubjectProperties (encoding :: k) =
+    SubjectProperties Subject [PropertyDescription encoding]
     deriving (Show, Eq, Generic)
-    deriving anyclass (ToJSON, FromJSON)
 
-instance Pretty Property where
-    pretty = viaShow
+deriving anyclass instance ToJSON (SubjectProperties 'AesonEncoding)
+deriving anyclass instance FromJSON (SubjectProperties 'AesonEncoding)
 
-data HashFunction =
-    SHA256
-    deriving (Show, Eq, Ord, Generic)
+instance FromJSON (SubjectProperties 'ExternalEncoding) where
+    parseJSON =
+        withObject "SubjectProperties" $ \o -> do
+            subject <- o .: "subject"
+            SubjectProperties subject <$>
+                accumulateSuccesses
+                    [ parseName o
+                    , parseDescription o
+                    , parsePreimage o
+                    , parseOwner o
+                    ]
+
+-- | Accumulate the results from _every_ matching parser.
+accumulateSuccesses ::
```

Comment:
This seems like it ought to be a standard function of some kind...

Reply:
It has the same type signature as `sequence`, but different semantics. I couldn't find anything that matched. Maybe it should show up as something like`Control.Alternative.every`, but I couldn't find a match.